libs/@local/harpc/client/typescript/src/types/SubsystemDescriptor.ts

```diff
@@ -0,0 +1,106 @@
+import type { FastCheck } from "effect";
+import {
+  Effect,
+  Equal,
+  Function,
+  Hash,
+  Inspectable,
+  pipe,
+  Pipeable,
+  Predicate,
+} from "effect";
+
+import { createProto, encodeDual } from "../utils.js";
+import type * as Buffer from "../wire-protocol/Buffer.js";
+import * as SubsystemId from "./SubsystemId.js";
+import * as Version from "./Version.js";
+
+const TypeId: unique symbol = Symbol(
+  "@local/harpc-client/wire-protocol/types/SubsystemDescriptor",
+);
+export type TypeId = typeof TypeId;
+
+export interface SubsystemDescriptor
+  extends Equal.Equal,
+    Inspectable.Inspectable,
+    Pipeable.Pipeable {
+  readonly [TypeId]: TypeId;
+
+  readonly id: SubsystemId.SubsystemId;
+  readonly version: Version.Version;
+}
+
+const SubsystemDescriptorProto: Omit<SubsystemDescriptor, "id" | "version"> = {
```

Comment:
We recently renamed the `Service` trait to `Subsystem` instead as it is such an overloaded term, a `Subsystem` for example would encompass all of the current `/account` REST routes.

Reply:
Maybe we can discuss this sync at some point, it'd be good if it more clearly tied into other names so there's less mental translation required. These groupings are called `Resource`s in `hash-graph-api`, for example. I may be misunderstanding the mapping though.

Reply:
SpiceDB (or Zanzibar?) uses the term `resource`, we just reused that for permission request. We don't use `Resource` anywhere else. `Subsystem` here refers to the actual API group of the graph, such as account management, data type management, etc.. I find "Resource" misleading if it also manages accounts or account groups.

Reply:
The API routes are all grouped as `XResource`, no?

```rs
/// With REST, we define resources that can be acted on. These resources are defined through
/// routes and HTTP methods.
///
/// This trait encapsulates a way to define related resource operations
/// through a `Router`, making it explicitly clear we want to provide `OpenApi` specification as
/// documentation for the routes.
pub(crate) trait RoutedResource: utoipa::OpenApi {

impl RoutedResource for AccountResource {
    /// Create routes for interacting with accounts.
    
impl RoutedResource for DataTypeResource {
    /// Create routes for interacting with data types.
```

Reply:
This `Routes` trait is the reason why we had to create common types such as 
```
    type Store<'pool, A: AuthorizationApi>: AccountStore
        + DataTypeStore
        + PropertyTypeStore
        + EntityTypeStore
        + EntityStore
        + Send
        + Sync;
```
and
```
pub trait RestApiStore:
    AccountStore + DataTypeStore + PropertyTypeStore + EntityTypeStore + EntityStore + TypeFetcher
```

just to avoid repeating all these types everywhere (because of the enforced common interface) even if they are distinct otherwise.

Reply:
I'm not against calling something a 'subsystem', my main point is that it's unclear at the moment from the codebase what it means, because (a) there's nowhere we list subsystems (unless I missed it), and (b) the place where we _do_ give a name to the same groupings, it's `Resource`.

I happen to think that 'resource' is more immediately understandable because all these group names do map onto resources (albeit some of them are SpiceDB resources), but it's possible that they won't always in future.

What we do need is for someone to see the word 'subsystem' and be able to find out what it means, which I assume will come with further implementation.